src/src/AppHelper.cpp

```diff
@@ -38,9 +39,32 @@ bool LoadBuiltInShaders() {
     return false;
   }
 
+  const char shadowmapvs[] = R"(#version 300 es
+  precision mediump float;
+  precision highp sampler2DShadow;
+  #include "mineola_builtin_uniforms"
+  in vec3 Pos;
+  void main() {
+    vec4 pos = _model_mat * vec4(Pos, 1.0);
+    gl_Position = _light_proj_mat_0 * _light_view_mat_0 * pos;
+  }
+  )";
+
+  const char shadowmapps[] = R"(#version 300 es
+  precision mediump float;
+  void main() {
+  }
+  )";
+
+  if (!CreateEffectFromMemHelper("mineola:effect:shadowmap_fallback",
+    shadowmapvs, shadowmapps, nullptr, std::move(render_states))) {
+    return false;
+  }
+
   const char diffusevs[] =
   "#version 300 es\n"
   "precision mediump float;\n"
+  "precision highp sampler2DShadow;\n"
```

Comment:
is highp sampler necessary here?

Reply:
Yes, otherwise shaders won’t compile on iOS. medium sampler also works.